flloat/parser/ldlf.lark

```diff
@@ -1,56 +1,58 @@
 start: ldlf_formula
-ldlf_formula:  ldlf_equivalence
-
-ldlf_equivalence: ldlf_implication (EQUIVALENCE ldlf_implication)*
-ldlf_implication: ldlf_or (IMPLY ldlf_or)*
-ldlf_or: ldlf_and (OR ldlf_and)*
-ldlf_and: ldlf_box (AND ldlf_box)*
-ldlf_box: BOXLSEPARATOR regular_expression BOXRSEPARATOR ldlf_box
-        | ldlf_diamond
-ldlf_diamond: DIAMONDLSEPARATOR regular_expression DIAMONDRSEPARATOR ldlf_box
-            | ldlf_not
-ldlf_not: NOT* ldlf_atom
-ldlf_atom:   ldlf_tt
-           | ldlf_ff
-           | ldlf_last
-           | ldlf_end
-           | ldlf_wrapped
-ldlf_wrapped: LSEPARATOR ldlf_formula RSEPARATOR
-
-ldlf_tt.2: TT
-ldlf_ff.2: FF
-ldlf_last.2: LAST
-ldlf_end.2: END
-
-regular_expression: regular_expression_union
-
-regular_expression_union: regular_expression_sequence (UNION regular_expression_sequence)*
-regular_expression_sequence: regular_expression_star (SEQ regular_expression_star)*
-regular_expression_star: regular_expression_test STAR?
-                       | LSEPARATOR regular_expression RSEPARATOR STAR?
-regular_expression_test: regular_expression_propositional
-                       | ldlf_formula TEST
-                       | LSEPARATOR ldlf_formula RSEPARATOR TEST
-regular_expression_propositional: propositional_formula
-
-
-%import common.ESCAPED_STRING
-WHITESPACE: (" " | "\n")+
+
+?ldlf_formula:  ldlf_equivalence
+?ldlf_equivalence: ldlf_implication (EQUIVALENCE ldlf_implication)*
+?ldlf_implication: ldlf_or (IMPLY ldlf_or)*
+?ldlf_or: ldlf_and (OR ldlf_and)*
+?ldlf_and: ldlf_unaryop (AND ldlf_unaryop)*
+
+?ldlf_unaryop:     ldlf_box
+             |     ldlf_diamond
+             |     ldlf_not
+             |     ldlf_wrapped
+?ldlf_box: BOXLSEPARATOR regular_expression BOXRSEPARATOR ldlf_unaryop
+?ldlf_diamond: DIAMONDLSEPARATOR regular_expression DIAMONDRSEPARATOR ldlf_unaryop
+?ldlf_not: NOT ldlf_unaryop
+?ldlf_wrapped:    ldlf_atom
+             |    LSEPARATOR ldlf_formula RSEPARATOR
+?ldlf_atom:   ldlf_tt
+          |   ldlf_ff
+          |   ldlf_last
+          |   ldlf_end
+          |   prop_atom
+
+ldlf_tt: TT
+ldlf_ff: FF
+ldlf_last: LAST
+ldlf_end: END
+
+regular_expression: re_union
+
+?re_union:      re_sequence (UNION re_sequence)*
+?re_sequence:   re_star (SEQ re_star)*
+?re_star:       re_test STAR?
+?re_test:       TEST ldlf_formula
```

Comment:
Let's discuss together this point. 
Due to collisions in the lalr parser, I had to move the test operator to the left of the expression, instead of the right.
The main reason of this difficulty is: inside a regular expression, the parser has no way to guess whether the formula must be a propositional or a possibly temporal formula, until it sees the Test symbol `?`. However, proceeding from left to right, this is not a look-ahead of 1. Just by moving the symbol to the left, the issue is solved.

Why this did not happen before? Because, in the previous version, the set of atoms in LDLf were completely disjoint from the set of atoms in PL. They were {"tt", "ff", "true", "false"} for LDLf, and anything else for PL. However, the the set of LDLf atoms was way too strict.

I've looked for other alternatives, but all of them seem to require some sort of left delimiter.

Reply:
yes I like that, let's keep it to the left. It solves a lot of problems...